### PR TITLE
Fix Windows UEFI

### LINKED ISSUE #3473: Windows Install Detailed Steps
Cobbler 3.2.x and 3.3.3 Office Docs  Windows Install not very detailed
Related work according to the official documentation cannot be successfully installed Windows
Can you make a blog or practice tutorial to help install Windows more detailed in detail?
Thank you!!!

### PR DESCRIPTION
**Pull request by tpw56j**
## Linked Items

Fixes #3473

 

## Description

Fixes an issue with Windows UEFI + Docs + small fixes

## Behaviour changes

Old: 
 - BCD: Windows Boot Application (13200003)
 - BCD: Timeout: 30

New: 
- BCD: Windows Boot Loader
- BCD: add Path: \windows\system32\winload.exe
 - description of the sync_post_wingen.py trigger is documented
 - BCD Tiimeout: 0
 - fix: os_version must be in lowercase
 - fix: non-standard distro directory in distro_mirror
 - fix: path for winpe.wim in case of using wimboot

## Category

This is related to a:

- [X] Bugfix
- [x] Feature
- [ ] Packaging
- [X] Docs
- [ ] Code Quality
- [ ] Refactoring
- [ ] Miscellaneous

## Tests

- [ ] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [ ] No tests required

### CODE REVIEW

**Comment by SchoolGuy** [comment]
@tpw56j As the backports are already merged... Is this PR final or do you want to add something to it?

**Comment by tpw56j** [comment]
@SchoolGuy Thanks! This PR is not ready yet.

**Comment by tpw56j** [comment]
@SchoolGuy PR is ready for review.

**Comment by tpw56j** [comment]
@SchoolGuy I forgot to change the `utils.mkdir` test to check that it is recursive, as with its previous implementation through `os.makedirs`.

File: changelog.d/3476.added [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Windows BIOS PXE install (via syslinux pxelinux.0, linux.c32 and wimboot tftp/http)
+Windows BIOS PXE install (via grub2 grub.0 and wimboot tftp/http)
```
Comment (SchoolGuy) [inline changelog.d/3476.added:2]: According to my knowledge, we should only have a single line per changelog fragment, I think a comma-separated list inside the brackets is also okay. No need to split it over multiple lines.

File: changelog.d/3473.fixed [hunk 2]
```diff
@@ -0,0 +1 @@
+Windows UEFI
```
Comment (SchoolGuy) [inline changelog.d/3473.fixed:1]: Not very descriptive, I think giving a little more would be helpful.

File: changelog.d/3476.changed [hunk 3]
```diff
@@ -0,0 +1,5 @@
+os_version must be in lowercase
+path for winpe.wim in case of using wimboot
+BCD: Windows Boot Application (13200003) -> Windows Boot Loader \windows\system32\winload.exe
+BCD: Timeout: 30 -> 0
+Windows templates
```
Comment (SchoolGuy) [inline changelog.d/3476.changed:5]: Let's please try to summary this. There will be a link to the issue/PR and as such the people can always read the detailed commit messages then.

File: cobbler/modules/managers/import_signatures.py [hunk 4]
```diff
@@ -1015,6 +924,131 @@ def rsync_repo_adder(distribution: "Distro") -> None:
         """
         return
 
+    # ==========================================================================
+    # windows-specific
+
+    def import_winpe(self) -> None:
```
Comment (SchoolGuy) [inline cobbler/modules/managers/import_signatures.py:930]: Codacy says that this method is:

- quite large,
- has too many local variables,
- has too many statements and
- is too complex

If it is reasonable I would kindly request that we split this into multiple methods.

File: cobbler/modules/sync_post_wingen.py [hunk 5]
```diff
@@ -1,5 +1,43 @@
 """
-TODO
+Create Windows boot files
+
+To create Windows boot files, files are used that must be extracted from the distro. The ``cobbler import``"
+command extracts the required files and places them where the given trigger expects them to be found.
```
Comment (SchoolGuy) [inline cobbler/modules/sync_post_wingen.py:5]: Thank you for this very informative docstring! That is really helpful in my eyes!

File: docs/user-guide/wingen.rst [hunk 6]
```diff
@@ -1,392 +1,440 @@
...
-    Cobbler profile 1: pxeboot.001 → boot001.exe → 001 → wi001.wim → startnet.cmd → autounatten001.xml → post_install.cmd profile_name
-    ...
+    vi /etc/samba/smb.conf
+            [DISTRO]
+            path = /var/www/cobbler
+            guest ok = yes
+            browseable = yes
+            public = yes
+            writeable = no
+            printable = no
```
Comment (SchoolGuy) [inline docs/user-guide/wingen.rst:36]: Can we automate this step maybe somehow? I have never used samba before.

File: docs/user-guide/wingen.rst [hunk 7]
```diff
@@ -1,392 +1,440 @@
...
+.. code-block:: text
 
-Final steps
-===========
+    ExecStart=/usr/sbin/in.tftpd -s /var/lib/tftpboot
+        to:
+    ExecStart=/usr/sbin/in.tftpd -m /etc/tftpd.rules -s /var/lib/tftpboot
 
-- Restart the services:
+Restart the tftp:
```
Comment (SchoolGuy) [inline docs/user-guide/wingen.rst:342]:
```suggestion
Restart the tftp service:
```

**Review by SchoolGuy — COMMENTED** [review]
I think this is a very big step forward! Thanks for the contribution of this big integration. Hopefully, I get my cloud-init integration done soon, so we can release a new version.

In general this PR is fine but I left a few remarks I'd like to discuss.

File: changelog.d/3473.fixed (hunk 2 above)
Comment (tpw56j) [reply changelog.d/3473.fixed]: I redistributed the description more correctly between “added”, “changed” and “corrected”.

File: cobbler/modules/managers/import_signatures.py (hunk 4 above)
Comment (tpw56j) [reply cobbler/modules/managers/import_signatures.py:930]: I split this large method into a set of smaller ones and changed the logic a little.

File: docs/user-guide/wingen.rst (hunk 6 above)
Comment (tpw56j) [reply docs/user-guide/wingen.rst:36]: Yes, we can automate the creation of smb.conf in the same way as we did for dhcpd.conf. I'll try to do this in the next PR.

File: docs/user-guide/wingen.rst (hunk 7 above)
Comment (tpw56j) [reply docs/user-guide/wingen.rst]: Fixed

**Review by SchoolGuy — APPROVED** [review]
I think this PR is LGTM. I will clean the Git commit history tomorrow at work and then merge this PR if no objections from you or other contributions are raised.

**Comment by tpw56j** [comment]
@SchoolGuy I see that the PR has not yet been merged, maybe someone has objections?

**Comment by SchoolGuy** [comment]
@tpw56j For some obscure reason the import tests are now failing. It is the second attempt. I will have a look tomorrow.

**Comment by SchoolGuy** [comment]
I have no idea what GH actions is having trouble with. Since this is going fine with the same container image locally I am merging this.

```
47e42ddbb199:/code # make SYSTESTS='import-*' system-test
make -C system-tests
scripts/run-tests import-*

Patterns:
  import-*

Running tests:
grep: warning: * at start of expression
  import-debian-bookworm... ok
  import-debian-bullseye... ok
  import-freebsd-freebsd11.4... ok
  import-freebsd-freebsd12.1... ok
  import-freebsd-freebsd12.2... ok
  import-freebsd-freebsd12.3... ok
  import-freebsd-freebsd13.0... ok
  import-freebsd-freebsd13.1... ok
  import-freebsd-freebsd13.2... ok
  import-redhat-cloudlinux7... ok
  import-redhat-fedora36... ok
  import-redhat-fedora37... ok
  import-redhat-fedora38... ok
  import-redhat-rhel8... ok
  import-suse-opensuse15.0... ok
  import-suse-opensuse15.1... ok
  import-suse-opensuse15.2... ok
  import-suse-opensuse15.3... ok
  import-suse-opensuse15.4... ok
  import-suse-sles15sp2... ok
  import-suse-sles15sp3... ok
  import-suse-sles15sp4... ok
  import-ubuntu-bionic... ok
  import-ubuntu-focal... ok
  import-ubuntu-groovy... ok
  import-ubuntu-hirsute... ok
  import-ubuntu-impish... ok
  import-ubuntu-jammy... ok
  import-ubuntu-kinetic... ok
  import-ubuntu-lunar... ok
  import-ubuntu-trusty... ok
  import-ubuntu-xenial... ok
  import-vmware-esxi70... ok
  import-xen-xenserver720... ok
```

**Pull request merged**